metals/src/main/scala/scala/meta/internal/metals/ReferenceProvider.scala

```diff
@@ -94,7 +99,14 @@ final class ReferenceProvider(
             ReferencesResult.empty
         }
       case None =>
-        ReferencesResult.empty
+        // NOTE(olafur): we block here instead of returning a Future because it
+        // requires a significant refactoring to make the reference provider and
+        // its dependencies (including rename provider) asynchronous. The remote
+        // language server returns `Future.successful(None)` when it's disabled
+        // so this isn't even blocking for normal usage of Metals.
+        Await
+          .result(remote.references(params), Duration(1, TimeUnit.MINUTES))
```

Comment:
Shouldn't we have the timeout configured in MetalsServerConfig? Or at least keep it somewhere in a value like  `RemoteLanguageServer.defaultTimeout`